base_multi_image/models/owner.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+# © 2014 Serv. Tecnol. Avanzados (http://www.serviciosbaeza.com)
+#        Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
+# © 2015 Antiun Ingeniería S.L. - Jairo Llopis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import _, api, fields, models, tools
+
+
+class Owner(models.AbstractModel):
+    _name = "base_multi_image.owner"
+
+    image_ids = fields.One2many(
+        comodel_name='base_multi_image.image',
+        inverse_name='owner_id',
+        string='Images',
+        domain=lambda self: [("owner_model", "=", self._name)],
+        copy=True)
+    image_main = fields.Binary(
+        string="Main image",
+        store=False,
+        compute="_get_multi_image",
+        inverse="_set_multi_image_main")
+    image_main_medium = fields.Binary(
+        string="Medium image",
+        compute="_get_multi_image",
+        inverse="_set_multi_image_main_medium",
+        store=False)
+    image_main_small = fields.Binary(
+        string="Small image",
+        compute="_get_multi_image",
+        inverse="_set_multi_image_main_small",
+        store=False)
+
+    @api.multi
+    @api.depends('image_ids')
+    def _get_multi_image(self):
+        """Get a the main image for this object.
+
+        This is provided as a compatibility layer for submodels that already
+        had one image per record.
+        """
+        for s in self:
+            s.image_main = False
+            s.image_main_medium = False
+            s.image_main_small = False
+            if s.image_ids:
+                s.image_main = s.image_ids[0].image_main
+                s.image_main_medium = s.image_ids[0].image_medium
+                s.image_main_small = s.image_ids[0].image_small
```

Comment:
I dislike picking the first image you get, it is inflexible.
Also notice that the images model does not have an order by attribute.

I suggest adding a "sequence" field on images.
Then the user would be free to choose the presentation order and it would be OK for the first one to be the main image.